src/popup/components/Modals/MultisigVaultCreate.vue

```diff
@@ -308,18 +317,32 @@ export default defineComponent({
       currentStep.value = STEPS.form;
     }
 
-    function openReviewStep() {
+    async function openReviewStep() {
+      multisigCreationTx.value = await prepareMultisigCreationTx(
+        confirmationsRequired.value,
+        signers.value.map((signer) => signer.address),
+      );
+      multisigProgress.value = MULTISIG_CREATION_STEPS.preparing;
       currentStep.value = STEPS.review;
     }
 
     async function createMultisigAccount() {
+      const sdk = await getSdk();
       currentStep.value = STEPS.processing;
       try {
-        multisigAccount.value = await deployMultisigAccount(
-          confirmationsRequired.value,
-          signers.value.map((signer) => signer.address) as [string, string, ...string[]],
+        // Send transaction to the chain
+        const { txHash } = await sdk.api.postTransaction(
+          { tx: multisigCreationTx.value.rawTx },
         );
-        signers.value = [];
+        await sdk.poll(txHash);
+        multisigProgress.value = MULTISIG_CREATION_STEPS.deployed;
+
+        const gaContract = await sdk.getAccount(multisigCreationTx.value.accountId);
+        multisigProgress.value = MULTISIG_CREATION_STEPS.created;
+        multisigAccount.value = {
+          contractId: gaContract.contractId,
+          multisigAccountId: multisigCreationTx.value.accountId,
+        };
```

Comment:
In my opinion this logic should be moved into the `multisigAccounts` composable.